### PR TITLE
[v6r22] Check if MQ is setup for Monitoring

### LINKED ISSUE #4446: Errors when using monitoring without a MQ
The MonitoringReporter has two failover options:
https://github.com/DIRACGrid/DIRAC/blob/b2ad824a059ee92922f7c585ca6e203d91f18842/MonitoringSystem/Client/MonitoringReporter.py#L2-L8

So one can use the monitoring without an MQ, but in this case one gets errors in the log

```
2020-02-11 13:15:24 UTC WorkloadManagement/StatesMonitoringAgent/getConfigurationTree ERROR: getOptionsDict() failed with message: Path /Resources/MQServices does not exist or it's not a section
2020-02-11 13:15:24 UTC WorkloadManagement/StatesMonitoringAgent ERROR: Failed to setupConnection: Requested destination not found in the CS: Monitoring::Queues::dirac.wmshistory
2020-02-11 13:15:24 UTC WorkloadManagement/StatesMonitoringAgent WARN: Fail to create Producer: Requested destination not found in the CS: Monitoring::Queues::dirac.wmshistory
```

which in my opinion is not correct, because MQ is not an absolute requirement for WMS Monitoring. They could be warnings, but certainly not errors in this case.

The problem is here in this call to `createProducer` inside `__createProducer`
https://github.com/DIRACGrid/DIRAC/blob/b2ad824a059ee92922f7c585ca6e203d91f18842/MonitoringSystem/Client/MonitoringReporter.py#L177-L189
this then somewhere deep down calls this
https://github.com/DIRACGrid/DIRAC/blob/5c9c1c8da4c377c8623f6154a5ca531d136ec671/Resources/MessageQueue/Utilities.py#L31
and produces an error inside the getConfigurationTree

I could add here a CS variable to check if a MQ is expected or not, but is this really the best solution?

### PR DESCRIPTION
**Pull request by petricm**
Resolves #4446
BEGINRELEASENOTES
*ConfigurationSystem
FIX: remove logging from inside getConfigurationTree
*MonitoringSystem
FIX: add check if MQ is setup for Monitoring

ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
These log messages have probably been added for a reason, I don't think they should be removed.

**Comment by petricm** [comment]
It's from here https://github.com/DIRACGrid/DIRAC/commit/d1918b0908e4488b0010bcecb8041d6b6783983b for some GOCDB2CSAgent reason. But I don't think such utility calls should log, what if I would just want to check if a section exists?

**Comment by petricm** [comment]
If there is no log inside I could use the same call
```py
result = gConfig.getConfigurationTree('/Resources/MQServices', mqService, mqType, mqName)
```
to check in the `MonitoringReporter` if a MQ setup exists:)

**Comment by petricm** [comment]
And it's also the only function inside `ConfigurationClient` that logs, which is a bit of non uniform behavior.

**Comment by fstagni** [comment]
Fine, but still this is not a good reason for removing these logging messages. If one day, developer X decides to (re-)add some log messages, we'll never remember that this shouldn't be done because we had this issue. This isn't the way to solve #4446

**Comment by petricm** [comment]
No of course this alone is not a solution to #4446, but I think it better to use 
```
gConfig.getConfigurationTree('/Resources/MQServices')
```
as a condition for determining if to use MQ or not, and we can avoid the introduction of an additional parameter into the CS.

**Comment by chaen** [comment]
I am with @petricm on that one. Low level utility classes should not log anything, except potentially debug. It is up to the caller to decide whether the logic is faulty or not, and log an error accordingly.

**Comment by petricm** [comment]
Is this acceptable?

**Comment by zmathe** [comment]
MonitoringReporter is using MQ for failover. If you do not have MQ, then in case of ES problem all data will be lost. In my opinion we should have MQ and we should not hide the error. 
@petricm  If you do not see this error message and do not read the instructions https://dirac.readthedocs.io/en/latest/AdministratorGuide/Systems/MonitoringSystem/index.html, how do you figure out that you need to use MQ for failover?

File: MonitoringSystem/Client/MonitoringReporter.py [hunk 1]
```diff
@@ -176,9 +181,11 @@ def __createProducer(self):
...
-      gLogger.warn("Fail to create Producer:", result['Message'])
-    else:
-      mqProducer = result['Value']
+    result = gConfig.getConfigurationTree('/Resources/MQServices')
+    if result['OK']:
+      result = createProducer("Monitoring::Queues::%s" % self.__failoverQueueName)
+      if not result['OK']:
+        gLogger.error("Fail to create Producer:", result['Message'])
+      else:
+        mqProducer = result['Value']
```
Comment (zmathe) [inline MonitoringSystem/Client/MonitoringReporter.py]: You should print a warning message.
```pyhton
    result = gConfig.getConfigurationTree('/Resources/MQServices')
    if result['OK']:
      result = createProducer("Monitoring::Queues::%s" % self.__failoverQueueName)
      if not result['OK']:
        gLogger.error("Fail to create Producer:", result['Message'])
      else:
        mqProducer = result['Value']
    else:
      gLogger.warn("Fail to create Producer:", result['Message'])
```

Comment (zmathe) [reply MonitoringSystem/Client/MonitoringReporter.py]: @petricm Could you please add a warning message?

**Comment by fstagni** [comment]
> MonitoringReporter is using MQ for failover. If you do not have MQ, then in case of ES problem all data will be lost. In my opinion we should have MQ and we should not hide the error.

MQ isn't (yet) strictly needed in DIRAC. So, the lack of MQ might be a warning, but can't be an error.

> @petricm If you do not see this error message and do not read the instructions https://dirac.readthedocs.io/en/latest/AdministratorGuide/Systems/MonitoringSystem/index.html, how do you figure out that you need to use MQ for failover?

Those instructions BTW would need a little update, e.g. RabbitMQ isn't the only solution (at least ActiveMQ is also possible).

**Pull request merged**